### ISSUE TITLE
"takeImage" not working

### ISSUE THREAD

**Issue by Torsten85** [servall/pi-camera-connect#1]
Hello,

the MJPEG "takeImage" part seems to be broken at the moment. The following code hangs forever:

```javascript
const { StreamCamera, Codec } = require("pi-camera-connect");
const fs = require("fs");

const runApp = async () => {
  const streamCamera = new StreamCamera({
    codec: Codec.MJPEG
  });


  await streamCamera.startCapture();
  const image = await streamCamera.takeImage();
  console.log('got image', image.length);

  await streamCamera.stopCapture();
};

runApp();
```

Seems like this sequence https://github.com/servall/pi-camera-connect/blob/master/src/lib/stream-camera.ts#L40 never occurs in the raspivid output.

Or maybe I'm missing something?

Thanks in advance

**Comment by Seikon** [comment]
Hi @Torsten85  I think the problem it´s basically  you can not take and image throught Stream camera, because stream camera only create a strem buffer for getting the data. Use this code instead if you want to go ahead with StreamCamera. If not, just use StillCamera.

const streamCamera = new StreamCamera({
    codec: Codec.H264
});

const videoStream = streamCamera.createStream();

await streamCamera.startCapture();

videoStream.on("data", data => console.log("New video data", data));

// Wait 5 seconds
await new Promise(resolve => setTimeout(() => resolve(), 5000));

await streamCamera.stopCapture();

Be aware of videoStream.on("data") doesn't bring the whole imge, it only provide the chunk binary data of the image.

**Comment by hitmaneac** [comment]
The jpegSignature in the streamCamera.js is wrong for takeImage, it's too long as there are similar sets of data elsewhere, it should be [0xFF, 0xD8, 0xFF, 0xDB, 0x00, 0x84, 0x00] instead of [0xFF, 0xD8, 0xFF, 0xE0, 0x00, 0x10, 0x4A, 0x46, 0x49, 0x46, 0x00].

**Comment by gregnr** [comment]
@Torsten85 Thanks for the issue and sorry for the delay! Could you please provide the Raspberry Pi hardware version, camera version (v1 or v2) and OS version you are using to reproduce this issue? I'm suspicious that the MJPEG signature is different based on the hardware.

**Comment by hitmaneac** [comment]
Hi, sorry for taking this long... So, the Pi version is 3b+, camera is 2.1, the OS is Raspbian GNU/Linux 9.8 (stretch).

**Comment by hitmaneac** [comment]
We should try and test this with as much different HW as possible, and dynamically select the correct jpegSignature when we have all the signatures. I'll try and get rpi v1 and v2 to check this, somehow I doubt it's the camera, I've tried it with the regular and the fisheye cam and both seem to work with my fix.

**Comment by raudette** [comment]
I applied the signature change suggested by hitmaneac  - it worked for me.  The PCB for my camera shows Rev 1.3

**Comment by gregnr** [comment]
This is now fixed in `v0.3.0`. Checks for Pi model version and provides the appropriate signature. Currently only supports Pi 3B/3B+/4B which all use the same signature, but I would like to add support for earlier Pi versions. PR's welcome!
